src/mscorlib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -203,6 +221,92 @@ public override void Return(T[] array, bool clearArray = false)
             }
         }
 
+        public bool Trim()
+        {
+            int milliseconds = Environment.TickCount;
+            MemoryPressure pressure = GetMemoryPressure();
+
+            ArrayPoolEventSource log = ArrayPoolEventSource.Log;
+            if (log.IsEnabled())
+                log.BufferTrimPoll(milliseconds, (int)pressure);
+
+            foreach (PerCoreLockedStacks bucket in _buckets)
+            {
+                bucket?.Trim((uint)milliseconds, Id, pressure, _bucketArraySizes);
+            }
+
+            if (pressure == MemoryPressure.High)
+            {
+                // Under high pressure, release all thread locals
+                foreach (KeyValuePair<T[][], object> tlsBuckets in s_AllTlsBuckets)
+                {
+                    T[][] buckets = tlsBuckets.Key;
+                    for (int i = 0; i < NumBuckets; i++)
```

Comment:
Why NumBuckets rather than buckets.Length?

Reply:
Using `buckets.Length` here would be faster (the bounds checks would be eliminated)

Reply:
Changing